cmsplugin_articles_ai/views.py

```diff
@@ -0,0 +1,73 @@
+# -*- coding: utf-8 -*-
+import json
+
+from django.conf import settings
+from django.utils.translation import ugettext_lazy as _
+from django.views.generic import DetailView, ListView
+
+from .models import Article, Tag, TagFilterMode
+
+
+class ArticleView(DetailView):
+    """
+    View for displaying single article.
+    """
+    model = Article
+    template_name = "cmsplugin_articles_ai/article_detail.html"
+    queryset = Article.objects.public()
+
+    def get_context_data(self, **kwargs):
+        context = super(ArticleView, self).get_context_data(**kwargs)
+        article = self.object
+        attachments = article.attachments.select_related("attachment_file")
+        context["image_attachments"] = attachments.images()
+        context["attachments"] = attachments.non_images()
+        context["tags"] = article.tags.select_related()
+        return context
+
+
+class ArticleListView(ListView):
+    """
+    View for listing all public articles or a list of public articles
+    per tag. Lists are paginated according to settings value.
+    """
+    context_object_name = "articles"
+    paginate_by = getattr(settings, "ARTICLES_PER_PAGE", 10)
+    tag_filter = ""
```

Comment:
This should be `None` too (for symmetry, as that's the default you declare in `get`).